syntax = "proto3";

service ShowMessage {
    rpc show(InputMessage) returns (OutputMessage) {}
}

message InputMessage {
    string value = 1;
}

message OutputMessage {
    string value = 1;
}

message TwitterCredentials {
    string consumer_key = 1;
    string consumer_secret = 2;
    string access_token = 3;
    string token_secret = 4;
}

message TwitterInputMessage {
    TwitterCredentials credentials = 1;
    string languages = 2;
    string sentences = 3;
    int64 time_limit = 4;
    int64 msg_limit = 5;
}

message TwitterOutputMessage {
    string value = 1;
}

service SentimentIntensityAnalysis {
    rpc intensivityAnalysis(InputMessage) returns (OutputMessage) {}
}

service SentimentComplexAnalysis {
    rpc complexAnalysis(InputMessage) returns (OutputMessage) {}
}

service CustomCorpusAnalysis {
    rpc customCorpusAnalysis(InputMessage) returns (OutputMessage) {}
}

service TwitterAnalysis {
    rpc twitterAnalysis(TwitterInputMessage) returns (OutputMessage) {}
}

service TwitterStreamAnalysis {
    rpc streamAnalysis(TwitterInputMessage) returns (OutputMessage) {}
}

service TrainData {
    rpc train(InputMessage) returns (OutputMessage) {}
}